saleor/warehouse/availability.py

```diff
@@ -246,9 +273,72 @@ def check_preorder_threshold_bulk(
         )
         for variant_id, channel_listings in variant_channels.items()
     }
+    return variants_channel_availability, variants_global_allocations
+
+
+def check_preorder_threshold(variant, quantity, channel_slug):
```

Comment:
As we spoke this function and the bulk version should have different names if they are used for different purposes.

Reply:
Done.